### PR TITLE
Relax methods visibility for Meta's internal classes' access

### PR DESCRIPTION
**Pull request by MnO2**
## Description

* Recent refactoring around HiveFilterPushdown has broken Meta's internal build. In order to fix it, we have done some refactoring and it requires the change of the visibility/staticness on a few base methods.

## Motivation and Context

* To fix the broken master for Meta's internal repo.

## Impact

* No impact to open source.

## Test Plan

* Unit tests

## Contributor checklist

- [ x ] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [ x ] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [ x ] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [ x ] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [ x ] Adequate tests were added if applicable.
- [ x ] CI passed.

## Release Notes

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-hive-common/src/main/java/com/facebook/presto/hive/rule/BaseSubfieldExtractionRewriter.java [hunk 1]
```diff
@@ -485,7 +485,7 @@ public Void visitVariableReference(
         }
     }
 
-    public static boolean useDynamicFilter(
```
Comment (jaystarshot) [inline presto-hive-common/src/main/java/com/facebook/presto/hive/rule/BaseSubfieldExtractionRewriter.java:488]: Why is this needed? Why can't the downstream dependency use this static method?

Comment (MnO2) [reply presto-hive-common/src/main/java/com/facebook/presto/hive/rule/BaseSubfieldExtractionRewriter.java:488]: We need to override this method. At this moment it is just a placeholder static method that always return false. With static method it is not dynamic binding in Java.

**Review by NikhilCollooru — APPROVED** [review]
